payload/src/main/java/io/jenkins/jenkinsfile/runner/Runner.java

```diff
@@ -106,6 +125,22 @@ public int run(PipelineRunOptions runOptions) throws Exception {
         return b.getResult().ordinal;
     }
 
+    private Folder createOrReturnFolder(Folder addToFolder, String folderName) {        
+        try {
+            Jenkins j = Jenkins.get();
+
+            if(addToFolder==null) {
+                Folder folder = j.getItem(folderName, j, Folder.class);
+                return  folder!=null ? folder : j.createProject(Folder.class, folderName);
+            }
+
+            Folder folder = j.getItem(folderName, addToFolder.getItemGroup(), Folder.class);
+            return folder !=null ? folder : addToFolder.createProject(Folder.class, folderName);
+        } catch (IOException ex) {
```

Comment:
Would be nice to add logging there. This is a wrong state which needs to be reported, and it should lead to the execution failure IMHO